components/world-4/breeding/arena.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+    Box,
+    Grid,
+    Text,
+} from 'grommet'
+import { useState, useContext, useMemo, useEffect } from "react";
+import { AppContext } from "../../../data/appContext";
+import { Breeding as BreedingDomain, petArenaBonuses, waveReqs } from "../../../data/domain/breeding";
+import IconImage from "../../base/IconImage";
+import ShadowBox from "../../base/ShadowBox";
+import { Player } from '../../../data/domain/player';
+import { ClassIndex, Talent } from '../../../data/domain/talents';
+import TextAndLabel from '../../base/TextAndLabel';
+import { TimeDisplaySize, TimeDown } from '../../base/TimeDisplay';
+
+export const ArenaBonusDisplay = () => {
+    const [breeding, setBreeding] = useState<BreedingDomain>();
+    const [playerData, setPlayerData] = useState<Player[]>();
+    const appContext = useContext(AppContext);
+
+    const beastMasters = useMemo(() => {
+        return playerData?.filter(player => (player.classId == ClassIndex.Beast_Master)) ?? [];
+    }, [playerData])
+
+    useEffect(() => {
+        if (appContext) {
+            const theData = appContext.data.getData();
+            setBreeding(theData.get("breeding"));
+            setPlayerData(theData.get("players"));
+        }
+    }, [appContext]);
+
+    if (!breeding) {
+        return (
+            <Box>
+                Still loading
+            </Box>
+        )
+    }
+    return (
+        <Box gap="small">
+            <Box direction="row" wrap justify="center" margin={{ top: 'small' }}>
+                {beastMasters && beastMasters.map((bm, index) => {
+                    const [arenaTalent, cooldown] = [...bm.cooldown.entries()].filter(([talent, cooldown]) => talent.skillIndex == 370)?.pop() as [Talent, number];
```

Comment:
I always hate hardcoding things like 370 in places like this. Is there an enum for this? If not, don't worry about it right now.

Reply:
Gonna check but I don't think there is one for talents, as when I made changes for the "Max Book level" checkbox in player talent pages all talents that had special treatment were hard coded too

Reply:
I don't think there is one, I generally went with specific hard-coded values since it made finding / comparing to game code easier.

But I do agree hardcoding these many values isn't ideal so probably a good time to create an enum for it if you are keen.

Reply:
There's a small list in domain/talents.tsx, I added this specific talent in it, it might be worth adding more to it later

Reply:
Or even creating a proper enum, potentially with "branching" so you could get the talent like this for example: talents.Beast_Master.ArenaSpirit so that you can avoid having all indexes of differents classes and specializations mixed together